.github/workflows/deploy.yml

```diff
@@ -0,0 +1,156 @@
+    steps:
+      - name: Define version
+        id: version
+        run: |
+          export COMMIT_SHA=${{ github.sha }}
+          export COMMIT_SHA_SHORT=${COMMIT_SHA:0:8}
+          export REF_NAME=${{ github.ref_name }}
+          export REF_SLUG=${REF_NAME//\//_}
+          if [[ ${REF_SLUG} == "master" ]]
+          then
+            VERSION=${REF_SLUG}-${COMMIT_SHA_SHORT}
+            echo "VERSION=${REF_SLUG}-${COMMIT_SHA_SHORT}" >> $GITHUB_OUTPUT
+          else
+            VERSION=${REF_SLUG}
+            echo "VERSION=${REF_SLUG}" >> $GITHUB_OUTPUT
+          fi
+          echo "set VERSION=${VERSION}"
+
+  build_push_docker:
+    name: Build docker image
+    runs-on: ubuntu-latest
+    environment: master_n_tags
+    needs: [set-variables]
+    env:
+      VERSION: ${{ needs.set-variables.outputs.VERSION }}
+    steps:
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+      - name: Build Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ./Dockerfile
+          push: true
+          tags: |
+            docker.io/parity/substrate-api-sidecar:${{ env.VERSION }}
+
+  deploy-stg-kusama:
+    name: Deploy Staging
+    runs-on: ubuntu-latest
+    needs: [set-variables, build_push_docker]
+    environment: parity-stg
+    env:
+      VERSION: ${{ needs.set-variables.outputs.VERSION }}
+      ARGOCD_SERVER: "argocd-stg.teleport.parity.io"
+    steps:
+      - name: Deploy to ArgoCD
+        uses: paritytech/argocd-deployment-action@main
+        with:
+          environment: "parity-stg"
+          tag: "${{ env.VERSION }}"
+          app_name: "substrate-api-sidecar-kusama"
+          app_packages: "${{ env.APP }}"
+          argocd_server: ${{ env.ARGOCD_SERVER }}
+          teleport_token: ${{ env.APP }}
+          teleport_app_name: "argocd-stg"
+          argocd_auth_token: ${{ secrets.ARGOCD_AUTH_TOKEN }}
+
+  deploy-stg-polkadot:
+    name: Deploy Staging
+    runs-on: ubuntu-latest
+    needs: [set-variables, build_push_docker]
+    environment: parity-stg
+    env:
+      VERSION: ${{ needs.set-variables.outputs.VERSION }}
+      ARGOCD_SERVER: "argocd-stg.teleport.parity.io"
+    steps:
+      - name: Deploy to ArgoCD
+        uses: paritytech/argocd-deployment-action@main
+        with:
+          environment: "parity-stg"
+          tag: "${{ env.VERSION }}"
+          app_name: "substrate-api-sidecar-polkadot"
+          app_packages: "${{ env.APP }}"
+          argocd_server: ${{ env.ARGOCD_SERVER }}
+          teleport_token: ${{ env.APP }}
+          teleport_app_name: "argocd-stg"
+          argocd_auth_token: ${{ secrets.ARGOCD_AUTH_TOKEN }}
+
+  deploy-prod-kusama:
+    name: Deploy Production
+    runs-on: ubuntu-latest
+    needs: [set-variables, deploy-stg-kusama, deploy-stg-polkadot]
```

Comment:
perhaps only stg-kusama needed. not sure though

Reply:
Thanks, makes sense.